frontend/src/Components/Link/index.tsx

```diff
@@ -0,0 +1,92 @@
+import React, { useEffect, useContext } from "react";
+import { usePlaidLink } from "react-plaid-link";
+import Button from "plaid-threads/Button";
+
+import Context from "../../Context";
+
+interface Props {
+  currentPath: string;
+}
+
+const Link: React.FC<Props> = (props: Props) => {
+  const {
+    itemId,
+    accessToken,
+    linkToken,
+    linkSuccess,
+    isItemAccess,
+    dispatch,
+  } = useContext(Context);
+
+  const onSuccess = React.useCallback(
+    (public_token: string) => {
+      // send public_token to server
+      const setToken = async () => {
+        const response = await fetch("/api/set_access_token", {
+          method: "POST",
+          headers: {
+            "Content-Type": "application/x-www-form-urlencoded;charset=UTF-8",
+          },
+          body: `public_token=${public_token}`,
+        });
+        if (!response.ok) {
+          dispatch({
+            type: "SET_STATE",
+            state: { itemId: `no item_id retrieved` },
+          });
+          dispatch({
+            type: "SET_STATE",
+            state: { accessToken: `no access_token retrieved` },
+          });
+          dispatch({ type: "SET_STATE", state: { isItemAccess: false } });
```

Comment:
here and in line 45/46 could you just use a single dispatch?

Reply:
done